src/main/java/sk/uniba/fmph/dcs/game_board/RewardMenu.java

```diff
@@ -0,0 +1,70 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import sk.uniba.fmph.dcs.stone_age.HasAction;
+import sk.uniba.fmph.dcs.stone_age.InterfaceTakeReward;
+import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RewardMenu implements InterfaceTakeReward {
+    private ArrayList<Effect> items;
+    private final Map<PlayerOrder, Player> players;
+
+    public RewardMenu(final ArrayList<Player> players) {
+        this.players = mapInitiate(players);
+    }
+
+    public final void initiate(final ArrayList<Effect> items) {
+        this.items = new ArrayList<>(items);
+
+    }

```

Comment:
```suggestion
    public final void initiate(final ArrayList<Effect> items) {
        this.items = new ArrayList<>(items);
        this.players = mapInitiate(allPlayers);
    }

```
treba vzdy na novo inicializovat players aby mohla existovat jedna instancia tejto treidy v celej hre